src/Parallel/MPIBasic.h

```diff
@@ -54,9 +54,15 @@ class MPIBasic
 	/** This rank */
 	int m_rank;
 
+	/** This rank */
+	int m_sharedMemMpiRank;
+
 	/** Number of processors */
 	int m_size;
 
+	/** Number of processors */
```

Comment:
Consider updating:
number of ranks in the shared memory sub-communicator

Reply:
done